### PR TITLE
Feature/correction

### PR DESCRIPTION
**Pull request by Amirmohsenjalili**
Please check for updates

### CODE REVIEW

File: src/Components/AppBar/index.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { styled } from '@mui/material/styles';
+import MuiAppBar, { AppBarProps as MuiAppBarProps } from '@mui/material/AppBar';
+import { drawerWidth } from '../../Components/Atoms';
+
+interface AppBarProps extends MuiAppBarProps {
+    open?: boolean;
+  }
+
+export const AppBar = styled(MuiAppBar, {
```
Comment (majidAbt) [inline src/Components/AppBar/index.ts]: Appbar is not a reusable component

File: src/Components/Atoms/index.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { atom } from 'jotai';
+
```
Comment (majidAbt) [inline src/Components/Atoms/index.ts]: atom is not a reusable component

File: src/Components/Main/index.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { styled } from '@mui/material/styles';
+
+export const Main = styled('main', { shouldForwardProp: (prop) => prop !== 'open' })<{
```
Comment (majidAbt) [inline src/Components/Main/index.ts]: main is not a reusable component

File: src/Components/UsersCard/index.tsx [hunk 4]
```diff
@@ -4,12 +4,13 @@ import CardMedia from '@mui/material/CardMedia';
...
 import { Box, CardActionArea } from '@mui/material';
 import { UserDataType } from './types';
+import styles from './style';
 
 const UserCard = ({ photoUrl, surName, firstName }:UserDataType) => {
 
   return (
     <Box display="inline-flex" flexWrap="wrap" justifyContent="center">
-      <Card sx={{ maxWidth: 345, m: 2}}>
+      <Card sx={styles.myCard}>
```
Comment (majidAbt) [inline src/Components/UsersCard/index.tsx]: for two sx args does not necessary to split it in style.ts file

File: src/Layout/index.tsx [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
-  let open: boolean = false;
+
   return (
-    <Box sx={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
-      <Header/>
-      <Main open={open}>
+    <Box sx={styles.myBox}>
+      <Header />
+      <Sidebar />
+      <Outlet />
```
Comment (majidAbt) [inline src/Layout/index.tsx]: why children and outlet both of these used here ??

File: src/Layout/Header/index.tsx [hunk 6]
```diff
@@ -104,53 +32,8 @@ export default function PersistentDrawerLeft() {
           <Typography  variant="h6" component="div" sx={{ flexGrow: 1 }}>
             Front end developers of Snap Grocery
           </Typography>
-          <Button color="inherit" onClick={() => navigate('/')}>Logout</Button>
+          <Button color="inherit" sx={style.myButton} onClick={() => navigate('/')}>Logout</Button>
```
Comment (majidAbt) [inline src/Layout/Header/index.tsx]: does not need style.ts

File: src/Layout/Header/style.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { SxProps } from '@mui/material';
+
+const myButton:SxProps = {
+  m: 2,
+  '&:hover': {
+    backgroundColor: '#90caf9',
```
Comment (majidAbt) [inline src/Layout/Header/style.ts]: does not need style anymore

File: src/Layout/Sidebar/index.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    justifyContent: 'flex-end',
+}));
+export default function Sidebar() {
+    
+const theme = useTheme();
+const [open, setOpen] = useAtom(DrawerOpenAtom);
+const handleDrawerClose = () => {
+  setOpen(false); 
+};
+const ListBar = ['List of Users', 'Create New User'];
```
Comment (majidAbt) [inline src/Layout/Sidebar/index.tsx]: const listBar = [ 
{
id: 1,
label: "List of Users",
value: "usersList",
},
{
id: 2,
label: "List of Users",
value: "usersList",
icon: "",
}
]

File: src/Pages/AddUser/index.tsx [hunk 9]
```diff
@@ -34,41 +55,38 @@ export default function InputFormProps() {
...
+                  placeholder={`Enter Your ${text}*`}  
+                />
+                {errors[text as keyof TFieldValues] && <Typography variant="caption" display="block" gutterBottom color='error'>{text} is required</Typography>}
+              </div>
+            ))}
+            <Button type="submit">Submit</Button>
+          </Stack>
+        </form>
+        {isError && error instanceof Error && <p>An error occurred: {error.message}</p>}
+        {isSuccess &&
```
Comment (majidAbt) [inline src/Pages/AddUser/index.tsx]: use notistack package instead of mui snackbar

Comment (Amirmohsenjalili) [reply src/Pages/AddUser/index.tsx]: I used notistack on the Signin page
that I have used every mode in the project

File: src/Layout/Layout.tsx [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    <Box
+      sx={{
+        display: "flex",
+        flexDirection: "column",
+        minHeight: "100vh",
+      }}
+    >
+      <Header />
+      <Sidebar />
+      <Main open={open}>
```
Comment (majidAbt) [inline src/Layout/Layout.tsx:22]: I see a Beautifull layout :D

File: package.json [hunk 11]
```diff
@@ -22,6 +22,7 @@
     "@types/react-dom": "^18.2.8",
```
Comment (majidAbt) [inline package.json]: check package.json . we must not have any types packages in dependencies, please move them to dev dependencies ;-)

**Review by majidAbt — APPROVED** [review]

**Pull request merged**